build-cluster-ec2.yml

```diff
@@ -74,30 +75,130 @@
+
+    - name: Add public ip to nodes
+      add_host:
+        groupname: "agent_nodes"
+        hostname: "{{ item.instances[0].public_ip }}"
+      with_items: "{{ ec2.results }}"
+
+    - name: Launch Public Proxy Nodes
+      local_action:
+        module: ec2
+        image: "{{ PROXY_IMAGE }}"
+        instance_type: "{{ PROXY_INSTANCE_TYPE }}"
+        keypair: "{{ KEYPAIR }}"
+        region: "{{ EC2_REGION }}"
+        group_id: "{{ PROXY_SECURITY_GROUP }}"
+        vpc_subnet_id: "{{ item }}"
+        assign_public_ip: yes
+        instance_tags:
+          Name: "{{ TAG_NAME }} Proxy"
+          Role: Proxy
+        count: 1
+        volumes:
+          - device_name: /dev/sda1
+            device_type: "{{ PROXY_VOL_TYPE }}"
+            volume_size: "{{ PROXY_VOL_SIZE }}"
+            delete_on_termination: true
+        wait: yes
+      register: ec2
+      with_items:
+        - "{{ SUBNET1 }}"
+        - "{{ SUBNET_SEED }}"
+
+    - name: Add public ip to nodes
+      add_host:
+        groupname: "proxy_nodes"
         hostname: "{{ item.instances[0].public_ip }}"
       with_items: "{{ ec2.results }}"
 
+    - name: Launch Template Nodes
+      local_action:
+        module: ec2
+        image: "{{ TEMPLATE_IMAGE }}"
+        instance_type: "{{ TEMPLATE_INSTANCE_TYPE }}"
+        keypair: "{{ KEYPAIR }}"
+        region: "{{ EC2_REGION }}"
+        group_id: "{{ AGENT_SECURITY_GROUP }}"
+        vpc_subnet_id: "{{ item }}"
+        assign_public_ip: yes
+        instance_tags:
+          Name: "{{ TAG_NAME }} Template"
+          Role: Template
+        count: 1
+        volumes:
+          - device_name: /dev/sda1
+            device_type: "{{ TEMPLATE_VOL_TYPE }}"
+            volume_size: "{{ TEMPLATE_VOL_SIZE }}"
+            delete_on_termination: true
+        wait: yes
+      register: ec2
+      with_items:
+        - "{{ SUBNET_SEED }}"
+
     - name: Wait for SSH to come up
       wait_for:
-        host: "{{ item.instances[0].public_dns_name }}"
-        port: 22
-        delay: 60
-        timeout: 320
-        state: started
+       host: "{{ item.instances[0].public_dns_name }}"
+       port: 22
+       delay: 60
+       timeout: 320
+       state: started
       with_items: "{{ ec2.results }}"
 
-- name: Setup ConductR
-  hosts: nodes
+- name: Setup Core Nodes
+  hosts: core_nodes
+  user: "{{ REMOTE_USER }}"
+  gather_facts: False
+  sudo: True
+
+  vars_files:
+    - "{{ VARS_FILE }}"
+
+  tasks:
+    - include: python/tasks/main.yml
+    - include: java/tasks/main.yml
+    - include: ntp/tasks/main.yml
+    - include: conductr/tasks/install-core.yml
+
+- name: Setup Private Nodes
+  hosts: agent_nodes
   user: "{{ REMOTE_USER }}"
   gather_facts: False
   sudo: True
 
   vars:
-# These are the default ConductR roles - these roles may be overridden within the VARS_FILE
+# These are the default ConductR Agent roles - these roles may be overridden within the VARS_FILE
     CONDUCTR_AGENT_ROLES:
```

Comment:
Can we rename `CONDUCTR_AGENT_ROLES` to `CONDUCTR_PRIVATE_AGENT_ROLES`?

Similarly the comment needs to be updated to `These are the default ConductR Private Agent roles - - these roles may be overridden within the VARS_FILE`

Once this is done, we can then modify https://github.com/typesafehub/conductr-ansible/pull/34/files#diff-a1aa4b9835ceeb0000cc881144e3762aR213

This is to allow user the ability to set different roles for private vs public nodes.